fetching/tarball-fetcher/src/remoteTarballFetcher.ts

```diff
@@ -158,27 +160,57 @@ export function createDownloader (
           ? throttle(opts.onProgress, 500)
           : undefined
         let downloaded = 0
+        const chunks: Buffer[] = []
         res.body!.on('data', (chunk: Buffer) => {
```

Comment:
Since you aren't wrapping any callback-based APIs other than the `data` event here, you might find it convenient to take advantage of the fact that Node's `Readable` interface implements `[Symbol.asyncIterator`:

```ts
// This will handle the 'data', 'error', and 'end' events.
for await (const chunk of res.body) {
  chunks.push(chunk);
  downloaded += chunk.length;
  onProgress?.(downloaded);
}

if (size !== null && size !== downloaded) {
  throw new BadTarballError(...);
}

// etc.
```